test/src/main/java/org/apache/accumulo/test/OfflineTableExtentCacheIT.java

```diff
@@ -0,0 +1,212 @@
+    SharedMiniClusterBase.getCluster().getClusterControl().start(ServerType.SCAN_SERVER,
+        "localhost");
+
+    String zooRoot = getCluster().getServerContext().getZooKeeperRoot();
+    ZooReaderWriter zrw = getCluster().getServerContext().getZooReaderWriter();
+    String scanServerRoot = zooRoot + Constants.ZSSERVERS;
+
+    while (zrw.getChildren(scanServerRoot).size() == 0) {
+      Thread.sleep(500);
+    }
+  }
+
+  @AfterAll
+  public static void stop() throws Exception {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  @Test
+  public void testScanWithCacheInvalidatedBySplit() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+
+      // Create table with no splits, ingest data, offline table
+      final int ingestedEntryCount =
+          ScanServerIT.createTableAndIngest(client, tableName, null, 10, 10, "colf");
+      client.tableOperations().offline(tableName, true);
+      assertFalse(client.tableOperations().isOnline(tableName));
+
+      int numInRange = 0;
+      // Scan table with single range in the middle of table
+      try (Scanner scanner = client.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(new Range("row_0000000005", "row_0000000006"));
+        scanner.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        numInRange = Iterables.size(scanner);
+        assertTrue(numInRange > 0 && numInRange <= ingestedEntryCount,
+            "The scan server scanner should have seen some ingested and flushed entries");
+      }
+
+      // Bring online, add splits, offline table
+      client.tableOperations().online(tableName, true);
+      assertTrue(client.tableOperations().isOnline(tableName));
+      SortedSet<Text> splits = new TreeSet<>();
+      splits.add(new Text("row_0000000001"));
+      splits.add(new Text("row_0000000002"));
+      splits.add(new Text("row_0000000003"));
+      splits.add(new Text("row_0000000004"));
+      splits.add(new Text("row_0000000005"));
+      splits.add(new Text("row_0000000006"));
+      splits.add(new Text("row_0000000007"));
+      splits.add(new Text("row_0000000008"));
+      splits.add(new Text("row_0000000009"));
+      client.tableOperations().addSplits(tableName, splits);
+      client.tableOperations().offline(tableName, true);
+      assertFalse(client.tableOperations().isOnline(tableName));
+
+      // Wait for SSERVER metadata cache to expire, set at 10s
+      Thread.sleep(15000);
+
+      // Re-scan
+      try (Scanner scanner = client.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(new Range("row_0000000005", "row_0000000006"));
+        scanner.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        assertEquals(numInRange, Iterables.size(scanner),
+            "The scan server scanner should have seen the same number of entries");
+      }
+
```

Comment:
These tests in #3091 invalidated the client cache at this location to be sure that the client looked up the new tablet information after the splits were added to the table. Can we be sure that the client code has this updated information here and is not using old information?